cmd/commands/globusCheckTransfer.go

```diff
@@ -0,0 +1,260 @@
+		skipDestPathCheck, _ := cmd.Flags().GetBool("skip-dest-path-check")
+		tapecopies, _ := cmd.Flags().GetInt("tapecopies")
+
+		if datasetUtils.TestFlags != nil {
+			datasetUtils.TestFlags(map[string]interface{}{
+				"testenv":              testenvFlag,
+				"devenv":               devenvFlag,
+				"localenv":             localenvFlag,
+				"tunnelenv":            tunnelenvFlag,
+				"user":                 userpass,
+				"token":                token,
+				"version":              showVersion,
+				"globus-cfg":           globusCfgFlag,
+				"mark-archivable":      globusCfgFlag,
+				"dry-run":              dryRun,
+				"autoarchive":          autoarchiveFlag,
+				"skip-dest-path-check": skipDestPathCheck,
+				"tapecopies":           tapecopies,
+			})
+			return
+		}
+
+		// === execute command ===
+
+		// show version
+		if showVersion {
+			fmt.Printf("%s\n", VERSION)
+			return
+		}
+
+		datasetUtils.CheckForNewVersion(client, "dummystring", VERSION)
+
+		// since Cobra doesn't support one way dependent flags, we have to do this:
+		if !markArchivable && (dryRun || autoarchiveFlag || tapecopies > 0) {
+			log.Fatalln("Can't use \"dry-run\", \"autoarchive\" or \"tapecopies\" if \"mark-archivable\" is not set.")
+		}
+		if !autoarchiveFlag && tapecopies > 0 {
+			log.Fatalln("Can't use \"tapecopies\" if \"autoarchive\" is not set.")
+		}
+
+		// find globus config
+		var globusConfigPath string
+		if cmd.Flags().Lookup("globus-cfg").Changed {
+			globusConfigPath = globusCfgFlag
+		} else {
+			execPath, err := os.Executable()
+			if err != nil {
+				log.Fatalln("can't find executable path:", err)
+			}
+			globusConfigPath = filepath.Join(filepath.Dir(execPath), "globus.yaml")
+		}
+
+		// environment overrides
+		if tunnelenvFlag {
+			APIServer = TUNNEL_API_SERVER
+		}
+		if localenvFlag {
+			APIServer = LOCAL_API_SERVER
+		}
+		if devenvFlag {
+			APIServer = DEV_API_SERVER
+		}
+		if testenvFlag {
+			APIServer = TEST_API_SERVER
+		}
+
+		// start message
+		startMessage := "Checking transfer complpetion"
+		if markArchivable {
+			startMessage += ", archivability"
+		}
+		if autoarchiveFlag {
+			startMessage += " and attempting auto-archival"
+		}
+		startMessage += " of the datasets corresponding to the following transfer tasks:\n"
+
+		color.Set(color.FgGreen)
+		log.Println(startMessage)
+		for _, task := range args {
+			fmt.Printf(" - %s\n", task)
+		}
+		color.Unset()
+
+		// logging into scicat and globus...
+		var user map[string]string
+		if markArchivable {
+			user, _ = authenticate(RealAuthenticator{}, client, APIServer, userpass, token)
+		}
+
+		globusClient, _, srcPrefixPath, _, _, err := cliutils.GlobusLogin(globusConfigPath)
+		if err != nil {
+			log.Fatalf("Couldn't create globus client: %v\n", err)
+		}
+
+		// go through each transfer task, and execute the requested operations
+		var archivableDatasetList []string
+		for _, taskId := range args {
+			task, err := globusClient.TransferGetTaskByID(taskId)
```

Comment:
so get taskById returns a list of tasks? I would expect it to return one

Reply:
What do you mean? It returns one task. I just loop over the list of tasks given as argument.

Reply:
ofc, my bad, sorry. Why would people need to check more than 1 globus task at a time though? And wouldn't they want to check by jobID rather than globus task?

Reply:
There's no archival job until we're certain that the files are on the PSI side, I can't use jobID for that. Unless you mean some other jobID, but I don't think there's one that fits the bill.

Reply:
ah right, I always get confused by that. So the user needs to get the list of taskIds somehow? Maybe it's worth printing the taskIds when the (globus) ingestion exists, so they can easily see them, without logging into globus. I also have the impression that the job should be created before the transfer, but not a strong opinion if users see the tasks list printed by the ingestor

Reply:
This would require us adding some sort of `hold` status to the job lifecycle so that scicat/nodered can postpone the creation actions until later. I think it increases the complexity.

Reply:
The task ID is printed out by GlobusTransfer(params) in cmd/cliutils/globusTransfer.go.